### ISSUE TITLE
Slow Jupyter Notebook Functioning in VS Code

### ISSUE THREAD

**Issue by david1111039243** [microsoft/vscode-jupyter#8320]
Having the same issue here. Running a jupyter notebook and it was working, now running incredibly slowly and only simple functions. I often have to quit out but similar issue happens after running a few cells in the notebook.

Visual Studio Code Version: 1.62.3 (Universal)
Mac OS 11.6 Big Sur
Python 3.9.2 64 bit
running Jupyter extension v2021.10.1101450599
python extension v2021.11.1422169775

Tried running print('hello world') and it worked, but took a long time (see screenshot 40+ seconds)
<img width="905" alt="Screen Shot 2021-11-20 at 7 19 00 PM" src="https://user-images.githubusercontent.com/43319024/142747807-218c6567-0514-4938-9347-5579f8b4031a.png">

Other more complex cells in book not running - it seems to time out mid way and give an 'x' icon (see screenshot)
<img width="905" alt="142744597-4572c102-141b-45ca-a85a-f11874b56c7e" src="https://user-images.githubusercontent.com/43319024/142747812-f2326881-8669-4391-bfd2-22fee3a476d5.png">

**Comment by peekxc** [comment]
Can confirm I am also facing this problem.

Info:
```
Version: 1.62.3
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T07:59:13.865Z (4 days ago)
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 19.6.0
```

A perhaps related thing is the use of the default MacOSX backend for interactive plotting. 

I've noticed that every *other* command input into the interactive Python terminal is ridiculously slow, possibly never finished until new input keystrokes are detected in the terminal. 

For example, on my machine a clean session results in the following: 

<img width="1164" alt="Screen Shot 2021-11-21 at 3 56 00 PM" src="https://user-images.githubusercontent.com/11429028/142778709-b1716ac1-7849-4302-bd94-84b92f6a65a1.png">

**Comment by DonJayamanne** [comment]
@peekxc please could you file a separate issue 
Also please could you install vscode insiders and then test and  provide the output from your Jupyter output panel
Thanks

**Comment by murphycj2** [comment]
I am also seeing the same issue, but specifically when running the jupyter notebook on a remote machine via the `Remote - SSH` package.

```
Version: 1.62.3 (Universal)
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T07:59:13.865Z (5 days ago)
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 20.6.0
```

Packages:
```
Remote - SSH : v0.66.1
Jupyter : v2021.10.1101450599
```

**Comment by david1111039243** [comment]
Ok I think this is the correct output but please let me know if I need to take other screenshots/modify the display. Thanks so much!
<img width="786" alt="Screen Shot 2021-11-22 at 1 25 49 PM" src="https://user-images.githubusercontent.com/43319024/142915348-405fe98b-9d6d-4b00-afea-84c27d594b28.png">

**Comment by DonJayamanne** [comment]
@david1111039243 @murphycj2 @peekxc 
Please could you file a new issue .
I think each of you are facing different issues, e.g. SSH, plots, and the like.

**Comment by DonJayamanne** [comment]
@murphycj2 
Please could you create a seprate issue, you are running on SSH and that could be unrelated. I'll look into that as well, but separtely, else it get a little confusing when we issues that are potentially unrelated

**Comment by DonJayamanne** [comment]
@david1111039243 @peekxc 
Please could you do the followign:
* Install VS Code Insiders from here https://code.visualstudio.com/insiders/
* Install the Jupyter extension (you will get the latest development version automatically, which contains a lot of fixes & perf improvements from our main branch)
* Please enable `verbose` logging as follows:
	* Go into settings (use the command `Preferences: Open setting UI`)
	* Search for the text `Jupyter logging` and select the value `verbose` as follows
![image](https://user-images.githubusercontent.com/1948812/142929343-8148d658-3bff-4a8c-a4f5-6e68f506a446.png)

* Next reload VS Code
* Replicate the issue you are running into (run just one cell)
* Go into the `Jupyter` output panel (use the command `Jupter: View Output`)
	* Copy everything (please copy all of the text) and paste it into this issue.
	*

**Comment by david1111039243** [comment]
Thanks for your help. This is what I'm getting. <img width="810" alt="Screen Shot 2021-11-22 at 11 08 04 PM" src="https://user-images.githubusercontent.com/43319024/142970038-7cec6e49-91e8-480d-ba19-f99fb9d6590f.png">
<img width="810" alt="Screen Shot 2021-11-22 at 11 10 24 PM" src="https://user-images.githubusercontent.com/43319024/142970113-357cbeed-539b-42cd-9c7e-f697e5380a16.png">

**Comment by DonJayamanne** [comment]
@david1111039243

* Are you using remote SSH
* are you running just one cell? Please run the cell only once, it looks like you are attempting to run it more than once.
* please run just one cell with a simple print statement.
* please could you copy the contents of the Jupyter output (instead of a screenshot)
* Please confirm you have enabled verbose logging

**Comment by david1111039243** [comment]
```
Verbose 18:32:13: IPyWidgetMessageDispatcher.initialize
Info 19:15:45: Execute Cells request 0
Info 19:15:45: Execute Cell 0 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 19:15:46: Cell 0 executed with state Success
```

This was the output, although the print statement worked. Let me try to re-create the issue. I am not using SSH

I believe I am using verbose logging - does that look correct?

**Comment by DonJayamanne** [comment]
> I believe I am using verbose logging - does that look correct?

Yes, however I need all of the output. Thanks again.
& thanks for your patience & getting back with the information.

**Comment by david1111039243** [comment]
Got it. So I re-created the issue, running a bit of code and then again running the print statement block. Cell 6 is the print statement. By the end, it ran but much slower than I would have expected...

I've pasted the full output below - at least I think that's what you were going for -if not, let me know what else to send. Thanks so much for your help troubleshooting!

`Jupyter Extension Version: 2021.11.1001493992.
Python Extension Verison: 2021.12.1482249642-dev.
Info 18:30:20: Experimentation service retrieved: [object Object]
User belongs to experiment group 'jupyterTest'
User belongs to experiment group 'jupyterEnhancedDataViewer'
Info 18:30:21: ZMQ install verified.
Verbose 18:30:21: Get Kernelspec root path, Class name = m (started execution), Arg 1: {"_isCancelled":false,"_emitter":null}
Verbose 18:30:21: Get Jupyter Paths, Class name = m (started execution), Arg 1: {"_isCancelled":false,"_emitter":null}
Verbose 18:30:21: Get Custom Env Variables, Class name = m (started execution), 
Verbose 18:30:21: Find kernel spec, Class name = x (started execution), Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: {"interpreter":{"hash":"aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"},"kernelspec":{"display_name":"Python 3.9.2 64-bit","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.9.2"},"orig_nbformat":4}, Arg 3: ""
Verbose 18:31:28: Get Custom Env Variables, Class name = m, completed in 67163ms, has a truthy return value, 
Verbose 18:31:28: Get Jupyter Paths, Class name = m, completed in 67166ms, has a truthy return value, Arg 1: {"_isCancelled":false,"_emitter":null}, Return Value: []
Verbose 18:31:28: Get Kernelspec root path, Class name = m, completed in 67169ms, has a truthy return value, Arg 1: {"_isCancelled":false,"_emitter":null}, Return Value: ["/usr/share/jupyter/kernels","/usr/local/share/jupyter/kernels","/Users/davidsperber/Library/Jupyter/kernels"]
Info 18:31:28: Experiment status for python is {"enabled":true,"optInto":[],"optOutFrom":[]}
Verbose 18:31:29: Getting Jupyter KernelSpec Root Path, Class name = m (started execution), 
Verbose 18:31:29: Get Kernelspec root path, Class name = m (started execution), 
Verbose 18:31:29: Get Jupyter Paths, Class name = m (started execution), Arg 1: undefined
Verbose 18:31:29: Get Kernelspec root path, Class name = m (started execution), Arg 1: {"_isCancelled":false,"_emitter":null}
Verbose 18:31:29: Get Jupyter Paths, Class name = m (started execution), Arg 1: {"_isCancelled":false,"_emitter":null}
Verbose 18:31:29: Getting Jupyter KernelSpec Root Path, Class name = m (started execution), 
Verbose 18:31:29: Get Kernelspec root path, Class name = m (started execution), 
Verbose 18:31:29: Get Jupyter Paths, Class name = m (started execution), Arg 1: undefined
Verbose 18:31:29: Getting Jupyter KernelSpec Root Path, Class name = m (started execution), 
Verbose 18:31:29: Get Kernelspec root path, Class name = m (started execution), 
Verbose 18:31:29: Get Jupyter Paths, Class name = m (started execution), Arg 1: undefined
Verbose 18:31:29: Getting Jupyter KernelSpec Root Path, Class name = m, completed in 7ms, has a truthy return value, , Return Value: "/Users/davidsperber/Library/Jupyter/kernels"
Verbose 18:31:29: Getting Jupyter KernelSpec Root Path, Class name = m, completed in 5ms, has a truthy return value, , Return Value: "/Users/davidsperber/Library/Jupyter/kernels"
Verbose 18:31:29: Getting Jupyter KernelSpec Root Path, Class name = m, completed in 4ms, has a truthy return value, , Return Value: "/Users/davidsperber/Library/Jupyter/kernels"
Verbose 18:31:29: Get Jupyter Paths, Class name = m, completed in 8ms, has a truthy return value, Arg 1: undefined, Return Value: []
Verbose 18:31:29: Get Jupyter Paths, Class name = m, completed in 7ms, has a truthy return value, Arg 1: {"_isCancelled":false,"_emitter":null}, Return Value: []
Verbose 18:31:29: Get Jupyter Paths, Class name = m, completed in 5ms, has a truthy return value, Arg 1: undefined, Return Value: []
Verbose 18:31:29: Get Jupyter Paths, Class name = m, completed in 5ms, has a truthy return value, Arg 1: undefined, Return Value: []
Verbose 18:31:29: Get Kernelspec root path, Class name = m, completed in 10ms, has a truthy return value, , Return Value: ["/usr/share/jupyter/kernels","/usr/local/share/jupyter/kernels","/Users/davidsperber/Library/Jupyter/kernels"]
Verbose 18:31:29: Get Kernelspec root path, Class name = m, completed in 10ms, has a truthy return value, Arg 1: {"_isCancelled":false,"_emitter":null}, Return Value: ["/usr/share/jupyter/kernels","/usr/local/share/jupyter/kernels","/Users/davidsperber/Library/Jupyter/kernels"]
Verbose 18:31:29: Get Kernelspec root path, Class name = m, completed in 8ms, has a truthy return value, , Return Value: ["/usr/share/jupyter/kernels","/usr/local/share/jupyter/kernels","/Users/davidsperber/Library/Jupyter/kernels"]
Verbose 18:31:29: Get Kernelspec root path, Class name = m, completed in 7ms, has a truthy return value, , Return Value: ["/usr/share/jupyter/kernels","/usr/local/share/jupyter/kernels","/Users/davidsperber/Library/Jupyter/kernels"]
Info 18:31:56: Setting setActiveController for ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Verbose 18:31:56: Intiailize notebook communications for editor ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Verbose 18:31:56: Resolving notebook UI Comms (resolve) for ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Verbose 18:31:56: initialize CommonMessageCoordinator
Verbose 18:31:57: Create activated Env, Class name = k (started execution), Arg 1: ""
Verbose 18:31:57: Getting activated env variables, Class name = O (started execution), Arg 1: undefined, Arg 2: "/usr/local/bin/python3", Arg 3: true
Info 18:31:57: KernelProvider switched kernel to id = .jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49./usr/local/bin/python3./usr/local/bin/python3.-m#ipykernel_launcher
Info 18:31:57: Starting Notebook in kernel.ts id = .jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49./usr/local/bin/python3./usr/local/bin/python3.-m#ipykernel_launcher
Info 18:31:57: Creating raw notebook for ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 18:31:57: Getting preferred kernel for ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 18:31:57: Computing working directory ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Verbose 18:31:57: Registering commtarget jupyter.widget
Verbose 18:31:57: IPyWidgetMessageDispatcher.initialize
Verbose 18:31:57: Created and initailized CommonMessageCoordinator
Verbose 18:31:57: Attach Coordinator for ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Verbose 18:31:57: Connecting to raw session for ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb with connection .jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49./usr/local/bin/python3./usr/local/bin/python3.-m#ipykernel_launcher
Info 18:31:57: Starting raw kernel Python 3.9.2 64-bit
Info 18:31:57: installMissingDependencies /usr/local/bin/python3
Verbose 18:31:57: Checking if product is installed, Class name = E (started execution), Arg 1: 19, Arg 2: "/usr/local/bin/python3"
Verbose 18:31:57: Create activated Env, Class name = k (started execution), Arg 1: ""
Verbose 18:31:57: Getting activated env variables, Class name = O (started execution), Arg 1: undefined, Arg 2: "/usr/local/bin/python3", Arg 3: true
Info 18:31:57: Find preferred kernel for ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb with metadata {"interpreter":{"hash":"aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"},"kernelspec":{"display_name":"Python 3.9.2 64-bit","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.9.2"},"orig_nbformat":4} & preferred interpreter /usr/local/bin/python3
Verbose 18:31:57: findPreferredKernel score for Python3, Python 3.9.2 64-bit is 116
Verbose 18:31:57: findPreferredKernel score for Python 3.9.2 64-bit is 116
Verbose 18:31:57: findPreferredKernel score for speclanguage=python, Python 2.7.16 64-bit is -1
Verbose 18:31:57: findPreferredKernel score for Python 2.7.16 64-bit is 1
Info 18:31:57: findKernel found Python 3.9.2 64-bit
Verbose 18:31:57: Find kernel spec, Class name = x, completed in 96234ms, has a truthy return value, Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: {"interpreter":{"hash":"aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"},"kernelspec":{"display_name":"Python 3.9.2 64-bit","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.9.2"},"orig_nbformat":4}, Arg 3: ""
Info 18:31:57: PreferredConnection: .jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49./usr/local/bin/python3./usr/local/bin/python3.-m#ipykernel_launcher found for NotebookDocument: ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 18:31:57: TargetController found ID: .jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49./usr/local/bin/python3./usr/local/bin/python3.-m#ipykernel_launcher for document ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 18:31:57: Setting controller affinity for ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb .jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49./usr/local/bin/python3./usr/local/bin/python3.-m#ipykernel_launcher
Info 18:31:57: Creating controller for jupyter-notebook with interpreter /usr/local/bin/python3
Verbose 18:31:58: Loading kernelspec from /Library/Frameworks/Python.framework/Versions/3.9/share/jupyter/kernels/python3/kernel.json for /usr/local/bin/python3
Info 18:31:59: Registering dummy command feature
Verbose 18:31:59: IPyWidgetMessageDispatcher.initialize
Verbose 18:32:09: IPyWidgetMessageDispatcher.initialize
Verbose 18:32:10: Getting activated env variables, Class name = O, completed in 13046ms, has a falsy return value, Arg 1: undefined, Arg 2: "/usr/local/bin/python3", Arg 3: true
Verbose 18:32:10: Creating execution process, Class name = k (started execution), Arg 1: {"pythonPath":"/usr/local/bin/python3"}
Verbose 18:32:10: Create ProcessService, Class name = h (started execution), Arg 1: undefined
Verbose 18:32:10: Get Custom Env Variables, Class name = m (started execution), Arg 1: undefined
Verbose 18:32:10: Cached data exists getEnvironmentVariables, <No Resource>
Verbose 18:32:10: Getting activated env variables, Class name = O, completed in 12353ms, has a falsy return value, Arg 1: undefined, Arg 2: "/usr/local/bin/python3", Arg 3: true
Verbose 18:32:10: Creating execution process, Class name = k (started execution), Arg 1: {"pythonPath":"/usr/local/bin/python3"}
Verbose 18:32:10: Create ProcessService, Class name = h (started execution), Arg 1: undefined
Verbose 18:32:10: Get Custom Env Variables, Class name = m (started execution), Arg 1: undefined
Verbose 18:32:10: Cached data exists getEnvironmentVariables, <No Resource>
Verbose 18:32:10: Get Custom Env Variables, Class name = m, completed in 5ms, has a truthy return value, Arg 1: undefined
Verbose 18:32:10: Get Custom Env Variables, Class name = m, completed in 8ms, has a truthy return value, Arg 1: undefined
Verbose 18:32:10: Create ProcessService, Class name = h, completed in 10ms, has a truthy return value, Arg 1: undefined, Return Value: {"_events":{},"_eventsCount":1,"decoder":{},"env":{"ELECTRON_RUN_AS_NODE":"1","USER":"davidsperber","__CFBundleIdentifier":"com.microsoft.VSCodeInsiders","COMMAND_MODE":"unix2003","LOGNAME":"davidsperber","SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.6hRb4225OC/Listeners","SHELL":"/bin/zsh","HOME":"/Users/davidsperber","__CF_USER_TEXT_ENCODING":"0x1F5:0x0:0x0","TMPDIR":"/var/folders/0n/z29k84cx7632vtls9bp1l2zr0000gn/T/","XPC_SERVICE_NAME":"application.com.microsoft.VSCodeInsiders.31008184.31008190","XPC_FLAGS":"0x0","ORIGINAL_XDG_CURRENT_DESKTOP":"undefined","VSCODE_CWD":"/","VSCODE_NLS_CONFIG":"{\"locale\":\"en-us\",\"availableLanguages\":{},\"_languagePackSupport\":true}","VSCODE_CODE_CACHE_PATH":"/Users/davidsperber/Library/Application Support/Code - Insiders/CachedData/4e303fcc52269ab578affa6cb1eb455eacab445f","VSCODE_IPC_HOOK":"/Users/davidsperber/Library/Application Support/Code - Insiders/1.63.0-insider-main.sock","VSCODE_PID":"24200","SHLVL":"0","PWD":"/","OLDPWD":"/","_":"/Applications/Visual Studio Code - Insiders.app/Contents/MacOS/Electron","VSCODE_AMD_ENTRYPOINT":"vs/workbench/services/extensions/node/extensionHostProcess","VSCODE_PIPE_LOGGING":"true","VSCODE_VERBOSE_LOGGING":"true","VSCODE_LOG_NATIVE":"false","VSCODE_IPC_HOOK_EXTHOST":"/var/folders/0n/z29k84cx7632vtls9bp1l2zr0000gn/T/vscode-ipc-c746a7e9-03a2-4701-ae1b-67faad4a65c0.sock","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_LOG_STACK":"true","APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL":"true","PATH":"/Library/Frameworks/Python.framework/Versions/3.9/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"},"processesToKill":{}}
Verbose 18:32:10: Create ProcessService, Class name = h, completed in 9ms, has a truthy return value, Arg 1: undefined, Return Value: {"_events":{},"_eventsCount":1,"decoder":{},"env":{"ELECTRON_RUN_AS_NODE":"1","USER":"davidsperber","__CFBundleIdentifier":"com.microsoft.VSCodeInsiders","COMMAND_MODE":"unix2003","LOGNAME":"davidsperber","SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.6hRb4225OC/Listeners","SHELL":"/bin/zsh","HOME":"/Users/davidsperber","__CF_USER_TEXT_ENCODING":"0x1F5:0x0:0x0","TMPDIR":"/var/folders/0n/z29k84cx7632vtls9bp1l2zr0000gn/T/","XPC_SERVICE_NAME":"application.com.microsoft.VSCodeInsiders.31008184.31008190","XPC_FLAGS":"0x0","ORIGINAL_XDG_CURRENT_DESKTOP":"undefined","VSCODE_CWD":"/","VSCODE_NLS_CONFIG":"{\"locale\":\"en-us\",\"availableLanguages\":{},\"_languagePackSupport\":true}","VSCODE_CODE_CACHE_PATH":"/Users/davidsperber/Library/Application Support/Code - Insiders/CachedData/4e303fcc52269ab578affa6cb1eb455eacab445f","VSCODE_IPC_HOOK":"/Users/davidsperber/Library/Application Support/Code - Insiders/1.63.0-insider-main.sock","VSCODE_PID":"24200","SHLVL":"0","PWD":"/","OLDPWD":"/","_":"/Applications/Visual Studio Code - Insiders.app/Contents/MacOS/Electron","VSCODE_AMD_ENTRYPOINT":"vs/workbench/services/extensions/node/extensionHostProcess","VSCODE_PIPE_LOGGING":"true","VSCODE_VERBOSE_LOGGING":"true","VSCODE_LOG_NATIVE":"false","VSCODE_IPC_HOOK_EXTHOST":"/var/folders/0n/z29k84cx7632vtls9bp1l2zr0000gn/T/vscode-ipc-c746a7e9-03a2-4701-ae1b-67faad4a65c0.sock","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_LOG_STACK":"true","APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL":"true","PATH":"/Library/Frameworks/Python.framework/Versions/3.9/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"},"processesToKill":{}}
Verbose 18:32:10: Creating execution process, Class name = k, completed in 14ms, has a truthy return value, Arg 1: {"pythonPath":"/usr/local/bin/python3"}, Return Value: {}
Verbose 18:32:10: Create activated Env, Class name = k, completed in 13064ms, has a truthy return value, Arg 1: ""
Info 18:32:10: Process Execution: > /usr/local/bin/python3 -m pip list
> /usr/local/bin/python3 -m pip list
Verbose 18:32:10: Creating execution process, Class name = k, completed in 20ms, has a truthy return value, Arg 1: {"pythonPath":"/usr/local/bin/python3"}, Return Value: {}
Verbose 18:32:10: Create activated Env, Class name = k, completed in 12375ms, has a truthy return value, Arg 1: ""
Info 18:32:10: Process Execution: > /usr/local/bin/python3 -c "import ipykernel"
> /usr/local/bin/python3 -c "import ipykernel"
Verbose 18:32:10: Kernel python392jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49 matches Python 3.9.2 64-bit based on interpreter path.
Verbose 18:32:10: Kernel python392jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49 matches Python 3.9.2 64-bit based on interpreter path.
Verbose 18:32:10: Kernel python392jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49 matches Python 3.9.2 64-bit based on interpreter path.
Verbose 18:32:11: Checking if product is installed, Class name = E, completed in 13904ms, has a truthy return value, Arg 1: 19, Arg 2: "/usr/local/bin/python3", Return Value: true
Verbose 18:32:11: Install Missing Dependencies, Class name = k, completed in 13908ms, has a falsy return value, Return Value: undefined
Info 18:32:11: Kernel launching with ports 9000,9001,9002,9003,9004. Start port is 9000
Verbose 18:32:11: Launching kernel in kernelProcess.ts, Class name = E (started execution), Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes", Arg 2: ""
Verbose 18:32:11: Launching kernel daemon, Class name = m (started execution), Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes", Arg 3: "python392jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49", Arg 4: "/usr/local/bin/python3"
Verbose 18:32:11: Get kernel daemon, Class name = y (started execution), Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: undefined, Arg 3: "/usr/local/bin/python3"
Verbose 18:32:11: Create daemon, Class name = k (started execution), Arg 1: {"daemonModule":"vscode_datascience_helpers.kernel_launcher_daemon","pythonPath":"/usr/local/bin/python3","dedicated":true,"resource":{"$mid":1,"fsPath":"/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb","external":"file:///Users/davidsperber/Documents/Coding/Learn%20Python%20with%20Baseball/Notebooks/My%20Notes/Full%20Count%20Nov%2020.ipynb","path":"/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb","scheme":"file"}}
Info 18:32:11: Launching kernel daemon for Python 3.9.2 64-bit # /usr/local/bin/python3
Info 18:32:11: No custom variables for Kernel as interpreter is not conda, but is Global
Verbose 18:32:11: Create activated Env, Class name = k (started execution), Arg 1: ""
Verbose 18:32:11: Getting activated env variables, Class name = O (started execution), Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: "/usr/local/bin/python3", Arg 3: true
Verbose 18:32:11: Getting activated env variables, Class name = O (started execution), Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: "/usr/local/bin/python3", Arg 3: true
Verbose 18:32:11: Getting activated env variables, Class name = O, completed in 74ms, has a falsy return value, Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: "/usr/local/bin/python3", Arg 3: true
Verbose 18:32:11: Creating execution process, Class name = k (started execution), Arg 1: {"resource":{"$mid":1,"fsPath":"/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb","external":"file:///Users/davidsperber/Documents/Coding/Learn%20Python%20with%20Baseball/Notebooks/My%20Notes/Full%20Count%20Nov%2020.ipynb","path":"/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb","scheme":"file"},"pythonPath":"/usr/local/bin/python3"}
Verbose 18:32:11: Create ProcessService, Class name = h (started execution), Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb"
Verbose 18:32:11: Get Custom Env Variables, Class name = m (started execution), Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb"
Verbose 18:32:11: Getting activated env variables, Class name = O, completed in 75ms, has a falsy return value, Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: "/usr/local/bin/python3", Arg 3: true
Verbose 18:32:11: Get Custom Env Variables, Class name = m, completed in 6ms, has a truthy return value, Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb"
Verbose 18:32:11: Create ProcessService, Class name = h, completed in 7ms, has a truthy return value, Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Return Value: {"_events":{},"_eventsCount":1,"decoder":{},"env":{"ELECTRON_RUN_AS_NODE":"1","USER":"davidsperber","__CFBundleIdentifier":"com.microsoft.VSCodeInsiders","COMMAND_MODE":"unix2003","LOGNAME":"davidsperber","SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.6hRb4225OC/Listeners","SHELL":"/bin/zsh","HOME":"/Users/davidsperber","__CF_USER_TEXT_ENCODING":"0x1F5:0x0:0x0","TMPDIR":"/var/folders/0n/z29k84cx7632vtls9bp1l2zr0000gn/T/","XPC_SERVICE_NAME":"application.com.microsoft.VSCodeInsiders.31008184.31008190","XPC_FLAGS":"0x0","ORIGINAL_XDG_CURRENT_DESKTOP":"undefined","VSCODE_CWD":"/","VSCODE_NLS_CONFIG":"{\"locale\":\"en-us\",\"availableLanguages\":{},\"_languagePackSupport\":true}","VSCODE_CODE_CACHE_PATH":"/Users/davidsperber/Library/Application Support/Code - Insiders/CachedData/4e303fcc52269ab578affa6cb1eb455eacab445f","VSCODE_IPC_HOOK":"/Users/davidsperber/Library/Application Support/Code - Insiders/1.63.0-insider-main.sock","VSCODE_PID":"24200","SHLVL":"0","PWD":"/","OLDPWD":"/","_":"/Applications/Visual Studio Code - Insiders.app/Contents/MacOS/Electron","VSCODE_AMD_ENTRYPOINT":"vs/workbench/services/extensions/node/extensionHostProcess","VSCODE_PIPE_LOGGING":"true","VSCODE_VERBOSE_LOGGING":"true","VSCODE_LOG_NATIVE":"false","VSCODE_IPC_HOOK_EXTHOST":"/var/folders/0n/z29k84cx7632vtls9bp1l2zr0000gn/T/vscode-ipc-c746a7e9-03a2-4701-ae1b-67faad4a65c0.sock","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_LOG_STACK":"true","APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL":"true","PATH":"/Library/Frameworks/Python.framework/Versions/3.9/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"},"processesToKill":{}}
Verbose 18:32:11: Creating execution process, Class name = k, completed in 8ms, has a truthy return value, Arg 1: {"resource":{"$mid":1,"fsPath":"/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb","external":"file:///Users/davidsperber/Documents/Coding/Learn%20Python%20with%20Baseball/Notebooks/My%20Notes/Full%20Count%20Nov%2020.ipynb","path":"/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb","scheme":"file"},"pythonPath":"/usr/local/bin/python3"}, Return Value: {}
Verbose 18:32:11: Create activated Env, Class name = k, completed in 83ms, has a truthy return value, Arg 1: ""
Info 18:32:11: Creating daemon process for /usr/local/bin/python3 with env variables count 0
Info 18:32:11: Process Execution: > /usr/local/bin/python3 -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> /usr/local/bin/python3 -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
Verbose 18:32:12: Create daemon, Class name = k, completed in 538ms, has a truthy return value, Arg 1: {"daemonModule":"vscode_datascience_helpers.kernel_launcher_daemon","pythonPath":"/usr/local/bin/python3","dedicated":true,"resource":{"$mid":1,"fsPath":"/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb","external":"file:///Users/davidsperber/Documents/Coding/Learn%20Python%20with%20Baseball/Notebooks/My%20Notes/Full%20Count%20Nov%2020.ipynb","path":"/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb","scheme":"file"}}
Verbose 18:32:12: Get kernel daemon, Class name = y, completed in 540ms, has a truthy return value, Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: undefined, Arg 3: "/usr/local/bin/python3"
Info 18:32:12: Starting kernel from scratch
Verbose 18:32:12: Python Daemon (pid: 24532): Execute rpc method exec_module in DS Daemon
Verbose 18:32:12: Python Daemon (pid: 24532): Execute rpc method exec_module from /usr/local/bin/python3
Verbose 18:32:12: Python Daemon (pid: 24532): Exec module in DS Kernel Launcher Daemon ipykernel_launcher with args ['--ip=127.0.0.1', '--stdin=9003', '--control=9001', '--hb=9000', '--Session.signature_scheme="hmac-sha256"', '--Session.key=b"eb7d03fa-82f0-4b3f-98bb-ff4deae857dc"', '--shell=9002', '--transport="tcp"', '--iopub=9004', '--f=/var/folders/0n/z29k84cx7632vtls9bp1l2zr0000gn/T/tmp-24276VMba47F1leRp.json']
Verbose 18:32:12: Python Daemon (pid: 24532): Exec in DS Kernel Launcher Daemon (observable) ipykernel_launcher with args ['--ip=127.0.0.1', '--stdin=9003', '--control=9001', '--hb=9000', '--Session.signature_scheme="hmac-sha256"', '--Session.key=b"eb7d03fa-82f0-4b3f-98bb-ff4deae857dc"', '--shell=9002', '--transport="tcp"', '--iopub=9004', '--f=/var/folders/0n/z29k84cx7632vtls9bp1l2zr0000gn/T/tmp-24276VMba47F1leRp.json']
Verbose 18:32:12: Python Daemon (pid: 24532): Exec in DS Kernel Launcher Daemon (observable) ['/usr/local/bin/python3', '-m', 'ipykernel_launcher', '--ip=127.0.0.1', '--stdin=9003', '--control=9001', '--hb=9000', '--Session.signature_scheme="hmac-sha256"', '--Session.key=b"eb7d03fa-82f0-4b3f-98bb-ff4deae857dc"', '--shell=9002', '--transport="tcp"', '--iopub=9004', '--f=/var/folders/0n/z29k84cx7632vtls9bp1l2zr0000gn/T/tmp-24276VMba47F1leRp.json']
Verbose 18:32:12: Python Daemon (pid: 24532): Exec in DS Kernel Launcher Daemon (observable)
Verbose 18:32:12: Python Daemon (pid: 24532): Kernel launched, with PID 24540
Verbose 18:32:12: Python Daemon (pid: 24532): Waiting for Kernel to die 24540
Verbose 18:32:12: Launching kernel daemon, Class name = m, completed in 551ms, has a truthy return value, Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb", Arg 2: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes", Arg 3: "python392jvsc74a57bd0aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49", Arg 4: "/usr/local/bin/python3", Return Value: <Return value cannot be serialized for logging>
Verbose 18:32:12: Launching kernel in kernelProcess.ts, Class name = E, completed in 553ms, has a truthy return value, Arg 1: "~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes", Arg 2: ""
Verbose 18:32:13: KernelProcess output: Content-Length: 371
Content-Type: application/vscode-jsonrpc; charset=utf8

{"jsonrpc": "2.0", "method": "log", "params": {"level": "INFO", "msg": "write to stderr: /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/traitlets/traitlets.py:2196: FutureWarning: Supporting extra quotes around Unicode is deprecated in traitlets 5.0. Use 'hmac-sha256' instead of '\"hmac-sha256\"' \u2013 or use CUnicode.\n", "pid": 24532}}
Verbose 18:32:13: Python Daemon (pid: 24532): write to stderr: /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/traitlets/traitlets.py:2196: FutureWarning: Supporting extra quotes around Unicode is deprecated in traitlets 5.0. Use 'hmac-sha256' instead of '"hmac-sha256"' – or use CUnicode.

Verbose 18:32:13: KernelProcess output: Content-Length: 346
Content-Type: application/vscode-jsonrpc; charset=utf8

{"jsonrpc": "2.0", "method": "output", "params": {"source": "stderr", "out": "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/traitlets/traitlets.py:2196: FutureWarning: Supporting extra quotes around Unicode is deprecated in traitlets 5.0. Use 'hmac-sha256' instead of '\"hmac-sha256\"' \u2013 or use CUnicode.\n"}}Content-Length: 115
Content-Type: application/vscode-jsonrpc; charset=utf8

{"jsonrpc": "2.0", "method": "log", "params": {"level": "INFO", "msg": "write to stderr:   warn(\n", "pid": 24532}}Content-Length: 90
Content-Type: application/vscode-jsonrpc; charset=utf8

{"jsonrpc": "2.0", "method": "output", "params": {"source": "stderr", "out": "  warn(\n"}}Content-Length: 397
Content-Type: application/vscode-jsonrpc; charset=utf8

{"jsonrpc": "2.0", "method": "log", "params": {"level": "INFO", "msg": "write to stderr: /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/traitlets/traitlets.py:2151: FutureWarning: Supporting extra quotes around Bytes is deprecated in traitlets 5.0. Use 'eb7d03fa-82f0-4b3f-98bb-ff4deae857dc' instead of 'b\"eb7d03fa-82f0-4b3f-98bb-ff4deae857dc\"'.\n", "pid": 24532}}Content-Length: 372
Content-Type: application/vscode-jsonrpc; charset=utf8

{"jsonrpc": "2.0", "method": "output", "params": {"source": "stderr", "out": "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/traitlets/traitlets.py:2151: FutureWarning: Supporting extra quotes around Bytes is deprecated in traitlets 5.0. Use 'eb7d03fa-82f0-4b3f-98bb-ff4deae857dc' instead of 'b\"eb7d03fa-82f0-4b3f-98bb-ff4deae857dc\"'.\n"}}Content-Length: 115
Content-Type: application/vscode-jsonrpc; charset=utf8

{"jsonrpc": "2.0", "method": "log", "params": {"level": "INFO", "msg": "write to stderr:   warn(\n", "pid": 24532}}Content-Length: 90
Content-Type: application/vscode-jsonrpc; charset=utf8

{"jsonrpc": "2.0", "method": "output", "params": {"source": "stderr", "out": "  warn(\n"}}
Warn 18:32:13: StdErr from Kernel Process /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/traitlets/traitlets.py:2196: FutureWarning: Supporting extra quotes around Unicode is deprecated in traitlets 5.0. Use 'hmac-sha256' instead of '"hmac-sha256"' – or use CUnicode.

Verbose 18:32:13: Python Daemon (pid: 24532): write to stderr:   warn(

Warn 18:32:13: StdErr from Kernel Process   warn(

Verbose 18:32:13: Python Daemon (pid: 24532): write to stderr: /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/traitlets/traitlets.py:2151: FutureWarning: Supporting extra quotes around Bytes is deprecated in traitlets 5.0. Use 'eb7d03fa-82f0-4b3f-98bb-ff4deae857dc' instead of 'b"eb7d03fa-82f0-4b3f-98bb-ff4deae857dc"'.

Warn 18:32:13: StdErr from Kernel Process /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/traitlets/traitlets.py:2151: FutureWarning: Supporting extra quotes around Bytes is deprecated in traitlets 5.0. Use 'eb7d03fa-82f0-4b3f-98bb-ff4deae857dc' instead of 'b"eb7d03fa-82f0-4b3f-98bb-ff4deae857dc"'.

Verbose 18:32:13: Python Daemon (pid: 24532): write to stderr:   warn(

Warn 18:32:13: StdErr from Kernel Process   warn(

Info 18:32:13: Raw session started and connected
Started kernel Python 3.9.2 64-bit
Info 18:32:13: Finished connecting 4871d3d7-ba96-401b-8c06-8399bee0f772
Info 18:32:13: UpdateWorkingDirectoryAndPath in Kernel
Info 18:32:13: changeDirectoryIfPossible
Info 18:32:13: Executing (status idle) silently Code = import os\nimport sys\n%cd "/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/
Info 18:32:13: Executing silently Code (completed) = import os\nimport sys\n%cd "/Users/davidsperber/Documents/Coding/Learn Python with Baseball/Notebooks/
Info 18:32:13: Executing (status idle) silently Code = %config Completer.use_jedi = False
Info 18:32:13: Executing silently Code (completed) = %config Completer.use_jedi = False
Info 18:32:13: Executing (status idle) silently Code = %config InlineBackend.figure_formats = ['png']
Info 18:32:13: Executing silently Code (completed) = %config InlineBackend.figure_formats = ['png']
Info 18:32:13: Executing (status idle) silently Code = import sys\nprint(sys.executable)
Info 18:32:13: Executing silently Code (completed) = import sys\nprint(sys.executable)
Info 18:32:13: Waiting for idle on (kernel): cb8c51b3-9293-4cde-bf2f-62d27f5f42b6 -> idle
Info 18:32:13: Finished waiting for idle on (kernel): cb8c51b3-9293-4cde-bf2f-62d27f5f42b6 -> idle
Verbose 18:32:13: IPyWidgetMessageDispatcher.initialize
Info 18:32:13: IPyWidgetScriptSource.initialize
Verbose 18:32:13: IPyWidgetMessageDispatcher.initialize
Verbose 18:32:13: IPyWidgetMessageDispatcher.initialize
Info 19:15:45: Execute Cells request 0
Info 19:15:45: Execute Cell 0 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 19:15:46: Cell 0 executed with state Success
Info 19:17:08: Execute Cells request 1
Info 19:17:08: Execute Cell 1 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 19:17:23: Cell 1 executed with state Success
Info 19:23:27: Execute Cells request 1
Info 19:23:27: Execute Cell 1 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 19:23:27: Cell 1 executed with state Success
Info 19:23:38: Execute Cells request 2
Info 19:23:38: Execute Cell 2 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 19:39:45: Cell 2 executed with state Success
Info 22:00:21: Execute Cells request 4
Info 22:00:21: Execute Cell 4 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 22:00:22: Cell 4 executed with state Success
Info 22:00:24: Execute Cells request 5
Info 22:00:24: Execute Cell 5 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 22:00:25: Cell 5 executed with state Success
Info 22:00:49: Execute Cells request 6
Info 22:00:49: Execute Cell 6 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 22:00:50: Execute Cells request 7
Info 22:00:50: Execute Cell 7 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 22:00:59: Cell 6 executed with state Success
Info 22:02:27: Cancel all remaining cells true || Error
Info 22:02:27: Cancel pending cells
Info 22:02:27: Cell 7 executed with state Error
Info 22:06:41: Execute Cells request 6
Info 22:06:41: Execute Cell 6 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 22:06:42: Cell 6 executed with state Success
Info 22:07:59: Execute Cells request 6
Info 22:07:59: Execute Cell 6 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 22:08:21: Execute Cells request 6
Info 22:08:21: Execute Cell 6 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 22:08:27: Execute Cells request 6
Info 22:08:27: Execute Cell 6 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 22:08:33: Cell 6 executed with state Success
Info 22:08:33: Cell 6 executed with state Success
Info 22:08:33: Cell 6 executed with state Success
Info 22:10:35: Execute Cells request 6
Info 22:10:35: Execute Cell 6 ~/Documents/Coding/Learn Python with Baseball/Notebooks/My Notes/Full Count Nov 20.ipynb
Info 22:10:59: Cell 6 executed with state Success
`

**Comment by david1111039243** [comment]
Hi there. Is that what you were looking for? Let me know if there is additional output you had in mind and I can send that over...

Also I get 'timeout' issue saying 'The window is not responding' pop up box, even without running code in the sheet. This is just after opening VSCode - Insiders, scrolling sheet and before any code is run. Interestingly, the markdown cells do not all load, or at least not immediately. Thanks so much for your help!

**Comment by DonJayamanne** [comment]
> This is just after opening VSCode - Insiders, scrolling sheet and before any code is run. Interestingly, the markdown cells do not all load, or at least not immediately. Thanks so much for your help!

Thanks @david1111039243 .
Transferring to VS Code, as it seems its stuck in loading the notebook or something else. Based on feedback, its stuck well before running a cell.

**Comment by david1111039243** [comment]
For what it's worth, I also run into issues with Jupyter labs trying to run this in my browser.  [Here is a post outlining what I'm](https://stackoverflow.com/questions/70148173/jupyter-notebooks-not-opening-in-web-browser) running into for Stack Overflow

**Comment by DonJayamanne** [comment]
Thanks for the response.
I'm going to close this issue, as it's not cause by vscode, ie if you don't use vscode you still run into this same issue.

**Comment by david1111039243** [comment]
Ok. So you think it is a Jupyter issue?

**Comment by DonJayamanne** [comment]
Yes, if you are running into this out side vscode and you have confirmed they you area running into the ranger issues in Jupyter lab, then head.

**Comment by morgankohler** [comment]
Has anyone found a solution to this problem?

**Comment by gusmaogabriels** [comment]
@morgankohler and @DonJayamanne, I am experiencing the same issue.
ipython notebooks running on a remote machine though vscode take over a minute to run a `print` command, whereas if I access the same running kernel by remote port forwarding a jupyterlab session to my localmachine the response is instantaneous.

**Comment by DmitriyG228** [comment]
Hi there, I am experiencing a very similar issue and obviously it is caused by vscode since the same notebook works perfectly well with jupyterlab. Please reopen the issue

**Comment by ogrisel** [comment]
I am experiencing a similar issue in VS Code when editing a Python file with the `# %%` cell markers to develop it interactively. Sometimes it works flawlessly (typically the first 5 cells execution after a kernel restart) and then for no apparent reason pressing "shift-enter" or clicking "Run cell" does not seem to have any effect in the jupyter side panel. Only after 10-30s or more, the side panel is updated. And then the next cell execution is slow again.

**Comment by DonJayamanne** [comment]
@morgankohler @DmitriyG228 @ogrisel apologies for missing the comments
please could you file a separate issue, with the relevant information.
